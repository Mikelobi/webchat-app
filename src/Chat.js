import { useEffect, useState } from 'react';
import MyMessage from './MyMessage';
import kio from './kio.jpg';

const Chat = ({ name }) => {
  const [allMessages, setAllMessages] = useState(['']);
  // eslint-disable-next-line
  useEffect(() => {
    setAllMessages(
      JSON.parse(window.localStorage.getItem('chat')),
      'from chat',
    );
  });
  return (
    <div className="chatBody">
      {allMessages.map((c, index) => {
        if (c.name === name) {
          return <MyMessage key={index} message={c.message} />;
        }
        return (
          <div key={index} className="avatar">
            <span>{c.message}</span>
            <span>
              <img src={kio} alt="user avater" />
            </span>
          </div>
        );
      })}
    </div>
  );
};
export default Chat;
