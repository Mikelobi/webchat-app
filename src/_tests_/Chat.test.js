import React from 'react';
import renderer from 'react-test-renderer';
import Chat from '../Chat';

it('renders correct', () => {
    const component = renderer.create(
      <Chat />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});