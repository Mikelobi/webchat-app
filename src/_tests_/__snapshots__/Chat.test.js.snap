// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct 1`] = `
<div
  className="chatBody"
>
  <div
    class="my-avatar"
  >
    <img
      alt="user avater"
      src="kio.jpg"
    />
    <span>
       
    </span>
  </div>
</div>
`;
